purchase_security/models/res_partner.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2024 Tecnativa - Víctor Martínez
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    # add indexes for better performance on record rules
+    user_id = fields.Many2one(index=True)
+
+    def _remove_key_followers(self, partner):
+        for record in self.mapped("commercial_partner_id"):
+            # Look for delivery and invoice addresses
+            childrens = record.child_ids.filtered(
+                lambda x: x.type in {"invoice", "delivery"}
+            )
+            (childrens + record).message_unsubscribe(partner_ids=partner.ids)
+
+    def _add_followers_from_salesmans(self):
+        """Sync followers in commercial partner + delivery/invoice contacts."""
```

Comment:
There's no delivery/invoice concept in purchases.

Reply:
You are right, it does not exist, but it is possible to select any partner in a purchase (Company > Delivery partner), that's why the followers are Sync in all the "children".

Reply:
No children syncing should be done in this case.